File: daySix/javaScript/FindTheDuplicatesInTheArray.js
function theDuplicatesIn(array) {
    let duplicateCounter = 0;
    let duplicatesArray = [];
    let thirdIndex = 0;
    for(let index = 0; index < array.length; index++){
        duplicateCounter = 1;
        for (let secondIndex = 0; secondIndex < array.length; secondIndex++){
            if(array[index] == array[secondIndex]) {
              duplicateCounter++;  
            }
        }
        if(duplicateCounter > 1){
            duplicatesArray.push(array[index]); 
        }
    }
    return duplicatesArray;
}

let numbers = [2,3,4,5,4,3,2,4,5]
console.log(theDuplicatesIn(numbers));

File: daySix/java/QuestionFour/FindTheDuplicatesInTheArray.java
import java.util.Arrays;
public class FindTheDuplicatesInTheArray {
    public static int countNumberOfDuplicatesIn(int [] numbers){
        int duplicateCounter = 0;
        int count = 0;
        for(int index = 0; index < numbers.length; index++){
            duplicateCounter = 1;
            for(int secondIndex = index + 1; secondIndex < numbers.length; secondIndex++){
                if(numbers[index] == numbers[secondIndex]) {
                    duplicateCounter++;
                }
            }
            if(duplicateCounter > 1) {
                count++;
            }
        }
        return count;
    }
    public static int[] theDuplicatesIn(int[] numbers){
        int duplicateCounter = 0;
        int duplicateArrayIndex = 0;
        int [] duplicatesArray = new int[countNumberOfDuplicatesIn(numbers)];
        for(int index = 0; index < numbers.length; index++){
            duplicateCounter = 1;
            for(int secondIndex = index + 1; secondIndex < numbers.length; secondIndex++){
                if(numbers[index] == numbers[secondIndex]) {
                    duplicateCounter++;
                }
            }
            if(duplicateCounter > 1) {
                duplicatesArray[duplicateArrayIndex] = numbers[index];
                duplicateArrayIndex++;
               
            } 
        }
        return duplicatesArray;
    }
}

